src/hapi/pipelines/database/wfp_market.py

```diff
@@ -0,0 +1,100 @@
+"""Populate the WFP market table."""
+
+from logging import getLogger
+from typing import Dict, List, Optional
+
+from hapi_schema.db_wfp_market import DBWFPMarket
+from hdx.location.adminlevel import AdminLevel
+from hdx.scraper.utilities.reader import Read
+from hdx.utilities.dictandlist import dict_of_dicts_add
+from sqlalchemy.orm import Session
+
+from ..utilities.logging_helpers import add_missing_value_message
+from . import admins
+from .base_uploader import BaseUploader
+
+logger = getLogger(__name__)
+
+
+class WFPMarket(BaseUploader):
+    def __init__(
+        self,
+        session: Session,
+        datasetinfo: Dict[str, str],
+        countryiso3s: List[str],
+        admins: admins.Admins,
+        adminone: AdminLevel,
+        admintwo: AdminLevel,
+    ):
+        super().__init__(session)
+        self._datasetinfo = datasetinfo
+        self._countryiso3s = countryiso3s
+        self._admins = admins
+        self._adminone = adminone
+        self._admintwo = admintwo
+        self.data = {}
+        self.name_to_code = {}
+
+    def populate(self):
+        logger.info("Populating WFP market table")
+        reader = Read.get_reader("hdx")
+        headers, iterator = reader.read(datasetinfo=self._datasetinfo)
+        warnings = set()
+        errors = set()
+        for market in iterator:
+            countryiso3 = market["countryiso3"]
+            if countryiso3 not in self._countryiso3s:
+                continue
+            name = market["market"]
+            adm1_name = market["admin1"]
+            if adm1_name is None:
+                add_missing_value_message(
+                    warnings, countryiso3, "admin 1 name for market", name
+                )
+                continue
```

Comment:
When admin1 name is missing, does that necessarily mean that the admin2 name is missing as well? I'm just wondering why you don't check admin2 first and use admin1 as a fall back

Reply:
There is only one case where admin 1 name is blank and in that case admin 2 name is also blank. I use the admin 1 name to get a pcode which I then use as parent when looking up the pcode corresponding to the admin 2 name (as it is possible for the same admin 2 name to exist in different countries or even in different admin 1s in the same country). There is a case where the admin 1 name is "Administrative unit not available" which I'll look at as part of the PR about improving the name matching: https://humanitarian.atlassian.net/browse/HDXDSYS-778.